doc.go

```diff
@@ -18,121 +18,235 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-// Package dig is the dig: Dependency Injection Framework for Go.
-//
-// package dig provides an opinionated way of resolving object dependencies.
+// Package dig provides an opinionated way of resolving object dependencies.
```

Comment:
Editing this file is in vein unless Makefile gets fixed and `gendoc` is removed.
Right now it's auto generated